thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/commons/ThirdEyeJiraClient.java

```diff
@@ -135,33 +161,82 @@ public void reopenIssue(Issue issue) {
   public void updateIssue(Issue issue, JiraEntity jiraEntity) {
     IssueInput issueInput = new IssueInputBuilder()
         .setAssigneeName(jiraEntity.getAssignee())
-        .setFieldInput(new FieldInput(IssueFieldId.LABELS_FIELD, jiraEntity.getLabels()))
-        // TODO: Will test this in the subsequent PR and then enable it
-        //.setFieldInput(new FieldInput(IssueFieldId.ATTACHMENT_FIELD, jiraEntity.getSnapshot()))
+        .setFieldInput(new FieldInput(PROP_LABELS, jiraEntity.getLabels()))
         .build();
 
     String prevAssignee = issue.getAssignee() == null ? "unassigned" : issue.getAssignee().getName();
     LOG.info("Updating Jira {} with [assignee={}, labels={}]. Previous state [assignee={}, labels={}]", issue.getKey(),
         jiraEntity.getAssignee(), jiraEntity.getLabels(), prevAssignee, issue.getLabels());
     restClient.getIssueClient().updateIssue(issue.getKey(), issueInput).claim();
+    restClient.getIssueClient().addAttachments(issue.getAttachmentsUri(), jiraEntity.getSnapshot()).claim();
+  }
+
+  Iterable<CimProject> getProjectMetadata(JiraEntity jiraEntity) {
+    return restClient.getIssueClient().getCreateIssueMetadata(new GetCreateIssueMetadataOptionsBuilder()
+        .withProjectKeys(jiraEntity.getJiraProject())
+        .withIssueTypeIds(jiraEntity.getJiraIssueTypeId())
+        .withExpandedIssueTypesFields()
+        .build()).claim();
+  }
+
+  private Map<String, CimFieldInfo> getIssueRequiredCreateFields(JiraEntity jiraEntity) {
+    Map<String, CimFieldInfo> requiredCreateFields = new HashMap<>();
+    CimProject project = getProjectMetadata(jiraEntity).iterator().next();
+    CimIssueType issueType = project.getIssueTypes().iterator().next();
```

Comment:
should we check `iterator().hasNext()` first?

Reply:
Added the check